src/Features/Core/Completion/MostRecentlyUsedList.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Completion
+{
+    internal class MostRecentlyUsedList<T>
+    {
+        private readonly object gate = new object();
```

Comment:
Naming rules use a leading underscore for private fields. i.e  _gate, _items and _maxSize